test/zksync-tests/AllowanceTransferTest.test.ts

```diff
@@ -0,0 +1,1162 @@
+import {MockERC20, Permit2} from "../../typechain-types";
+import {deployContract, provider} from "./shared/zkSyncUtils";
+import {Wallet} from "zksync-web3";
+import fs from "fs";
+import {BigNumber, BigNumberish, ethers} from "ethers";
+import {
+    AllowanceTransferDetails, getCompactPermitSignature, getPermitBatchSignature, getPermitSignature,
+    PermitBatch,
+    PermitSingle, TokenSpenderPair
+} from "./utils/PermitSignature";
+import {expect} from "./shared/expect";
+
+const RICH_WALLET_PRIVATE_KEYS = JSON.parse(fs.readFileSync("test/zksync-tests/shared/rich-wallets.json", 'utf8'));
+const DECIMAL_MULT: BigNumber = ethers.BigNumber.from(10).pow(ethers.BigNumber.from(18));
+
+describe("AllowanceTransferTest", function () {
+
+    const defaultExpiration: BigNumber = ethers.BigNumber.from(Date.now() + 50000);
+    let token0: MockERC20;
+    let token1: MockERC20;
+    let permit2: Permit2;
+    let from: Wallet;
+    let fromDirty: Wallet;
+    let spender: Wallet = new Wallet(RICH_WALLET_PRIVATE_KEYS[1].privateKey, provider);
+    let receiver: Wallet = new Wallet(RICH_WALLET_PRIVATE_KEYS[2].privateKey, provider);
+    let address3: Wallet = new Wallet(RICH_WALLET_PRIVATE_KEYS[3].privateKey, provider);
+
+    let defaultAmount: BigNumber = DECIMAL_MULT
+    let defaultNonce: BigNumberish = ethers.constants.Zero;
+    let dirtyNonce: BigNumberish = ethers.constants.One;
+    let MINT_AMOUNT_ERC20: BigNumber = defaultAmount.mul(10)
+
+    let fromPrivateKey: string;
+    let fromPrivateKeyDirty: string;
+    let blockTimestampDebug: BigNumberish;
+
+
+    beforeEach(async function () {
+
+        let l1TimeStamp: number = 0;
+        let l1BatchRange = await provider.getL1BatchBlockRange(
+            await provider.getL1BatchNumber()
+        );
+
+        if (l1BatchRange) {
+            l1TimeStamp = (await provider.getBlock(l1BatchRange[1])).l1BatchTimestamp;
+        }
+
+
+        blockTimestampDebug = ethers.BigNumber.from(l1TimeStamp + 80000000);
+
+        permit2 = <Permit2>await deployContract('Permit2');
+
+        fromPrivateKey = RICH_WALLET_PRIVATE_KEYS[0].privateKey;
+        fromPrivateKeyDirty = RICH_WALLET_PRIVATE_KEYS[5].privateKey;
+
+        from = new Wallet(fromPrivateKey, provider);
+        fromDirty = new Wallet(fromPrivateKeyDirty, provider);
+
+        token0 = <MockERC20>await deployContract('MockERC20', ["Test0", "TEST0", ethers.BigNumber.from(18)]);
+        token1 = <MockERC20>await deployContract('MockERC20', ["Test1", "TEST1", ethers.BigNumber.from(18)]);
+
+        await (await mint(from.address, from));
```

Comment:
@DmitriyIschenko What is the point of doing a double `await`? If nothing please remove it
```suggestion
        await mint(from.address, from);
```

Reply:
`.wait()` !!!